ls1intum/Artemis
# [Feature] Add Grading Instruction link for assessment feedback

### Checklist

### Motivation and Context



When the grading instruction is associated with feedback, there is no way to see this relation in UI. Also, the assessor can not remove this association directly in assessment view. To achieve this issue, simply `link ` icon is added to feedback block. When the assessor dropped the structured grading instruction into feedback, the link is appeared. When the user hovers over it, the small popup displays the associated grading instruction description. After one click,  `link ` icon changes into `trash ` icon and the popup warns user. After second click, the association between structured grading instruction and feedback is removed.

This PR only covers Text Block Feedback, Unreferenced Feedback and Inline Feedback for Programming Exercise. For UML element feedback, it requires some changes on Apollon. I'll make a PR for Apollon afterwards. 

** If you find translation problems, please correct me 😄 

### Steps for Testing


**Side not: If you want to test this for the programming exercise, you should activate the manual assessment. First you should set the `due date` for the exercise on `timeline`, then activate/click `Manual Review`.

1. Log in to Artemis
2. Navigate to Course Administration
3. Create/Find a new exercise.
4. Add an example solution for the exercise / participate this exercise as a student.
a. If you add an example submission, then create new assessment for this submission and use one of the structured grading instruction as a feedback.
b. If you participated this exercise as student, then login as tutor/instructor and assess this submission. During the assessment, use one of the structured grading instructions as a feedback.
5. You should see the link icon like in the following screenshots.
6. Mouse over it, check the popup. 
7. Click one time, see the delete icon and popup. Click again. link should be disappeared. 

### Test Coverage




.....

### Screenshots



Text-Block Feedback Editor For Text Exercise
<img src="https://user-images.githubusercontent.com/23650630/122055918-66c20f00-cde9-11eb-8613-6418b6149926.png" width="400" />

Text-Block Feedback Editor For Text Exercise
<img src="https://user-images.githubusercontent.com/23650630/122056007-7a6d7580-cde9-11eb-8f87-da04d5b75fd8.png" width="400" />

Unreferenced Feedback (Available for Text/File-Upload/Modeling/Programming Exercise)
<img src="https://user-images.githubusercontent.com/23650630/120950116-5e613880-c746-11eb-890d-e2b8c21b9446.png" width="400" />

Code Editor Inline Feedback For Programming Exercise
<img src="https://user-images.githubusercontent.com/23650630/121828751-c8d32500-ccc0-11eb-9117-3ab90cc2ed80.png" width="400" />
